app.py

```diff
@@ -326,19 +326,31 @@ def apply_tagging_method():
     else:
         docs = data["documents"]
         options = data["options"]
+
+
         auto_keywords = create_automated_keywords(docs, options["numClusters"], options["numKeywords"])
 
         doc_ids = auto_keywords.keys()
         docs = solr.docs.get(*doc_ids)
 
-        for doc in docs:
-            new_keywords = auto_keywords[doc.id]
-            doc.keywords.update(
+        if len(doc_ids)==1:
```

Comment:
Not a big fan of this if statement. Just do
`if len(doc_ids) == 1:
      doc_ids = [doc_ids]`

and then you dont have all the duplicate code.